compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2071,11 +2071,7 @@ class Typer extends Namer
     val parentsWithClass = ensureFirstTreeIsClass(parents.mapconserve(typedParent).filterConserve(!_.isEmpty), cdef.nameSpan)
     val parents1 = ensureConstrCall(cls, parentsWithClass)(using superCtx)
 
-    var forceEmptyBody: Boolean = false
-    val enumCaseDef = optEnumCase(cls)
-    if enumCaseDef.exists then
-      val firstParent = parents1.head.tpe.dealias.typeSymbol
-      forceEmptyBody = !checkEnum(cdef, cls, enumCaseDef, firstParent) // don't bother looking inside the template
+    @annotation.threadUnsafe lazy val firstParent = parents1.head.tpe.dealias.typeSymbol
```

Comment:
local lazy vals are always thread-unsafe so that annotation does nothing

Reply:
nice thanks